flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -332,9 +332,7 @@ getDependentTypeMemSizeFn(fir::RecordType recTy, fir::AllocaOp op,
                           mlir::ConversionPatternRewriter &rewriter) {
   auto module = op->getParentOfType<mlir::ModuleOp>();
   std::string name = recTy.getName().str() + "P.mem.size";
-  if (auto memSizeFunc = module.lookupSymbol<mlir::LLVM::LLVMFuncOp>(name))
-    return memSizeFunc;
-  TODO(op.getLoc(), "did not find allocation function");
+  return module.lookupSymbol<mlir::LLVM::LLVMFuncOp>(name);
```

Comment:
The work to support this is not completed, so this can and does fail on lookup. The TODO was a useful reminder that we need to finish the implementation.

Reply:
Ok, so is the mem size function supposed to be generated by this pass? If so then I agree that TODO is better. If it's supposed to already exist by the time this pass is run, then I think it's better to error out (and maybe put a TODO in whichever part of the compiler is expected to generate it). We weren't sure about the grand plan when this was reviewed in Phab :)

Reply:
> Ok, so is the mem size function supposed to be generated by this pass.

The detailed is design for F20XX is not completed yet. The idea behind the TODO here was that lowering would generate such function for derived type with length parameters, and that CodeGen would just find them and use them where it needs to. There is a TODO in lowering that won't allow derived type with length parameters to lower.
However, given the exact design here is not final at all, I think it's fair to also leave the TODO here.

Reply:
Ok, reinstated the TODO :)